src/Watchable.cpp

```diff
@@ -0,0 +1,50 @@
+#include "../include/Watchable.h"
+#include "../include/Session.h"
+
+
+Watchable::Watchable(long id, int length, const std::vector<std::string>& tags)
+: id(id), length(length), tags(tags){}
+
+std::string Watchable::getTagStrings() // auxillary to avoid code duplication
+{
+    std::string totalString = "[";
+    for(auto tag : tags)
+    {
+        totalString += tag;
+    }
+    return totalString+"]";
+}
+
+Movie::Movie(long id, const std::string& name, int length, const std::vector<std::string>& tags)
+: Watchable(id, length, tags), name(name)
+{}
+
+Watchable* Movie::getNextWatchable(Session& s)
+{
+    User& activeUser = s.getActiveUser();
+    return activeUser.getRecommendation(s);
+}
+
+std::string Movie::toString() const
+{
+    return name+" "+length+" minutes "+getTagsString(); // According to the "Inglorious Basterds 153 minutes [War, Western]" format
```

Comment:
I think formated is more readable
aka "%s %d minutes" + ...
check if possible here (:

Reply:
formatting a string outside of a printf (or similar) function is more work than needed. Plus the operator + is properly overloaded for strings, so why not use it. In my opinion it is also more readable because you can see what string is where and not just the fact that it is a string, but that is just preference. Either way I think formatting the string will take more work than is really needed

Reply:
OK. I accept, your code your style :)